src/control/cmd/daos_agent/fabric.go

```diff
@@ -273,6 +277,80 @@ func (n *NUMAFabric) setDefaultNUMANode() {
 	}
 }
 
+// Find finds a specific fabric device by name. There may be more than one domain associated.
+func (n *NUMAFabric) Find(name string) ([]*FabricInterface, error) {
+	if n == nil {
+		return nil, errors.New("nil NUMAFabric")
+	}
+
+	result := make([]*FabricInterface, 0)
+	for _, devs := range n.numaMap {
+		for _, fi := range devs {
+			if fi.Name == name {
+				result = append(result, copyFI(fi))
+			}
+		}
+	}
+
+	if len(result) > 0 {
+		return result, nil
+	}
+
+	return nil, fmt.Errorf("fabric interface %q not found", name)
+}
+
+// FindDevice looks up a fabric device with a given name, domain, and provider.
+// NB: The domain and provider are optional. All other parameters are required. If there is more
+// than one match, all of them are returned.
+func (n *NUMAFabric) FindDevice(params *FabricIfaceParams) ([]*FabricInterface, error) {
+	if params == nil {
+		return nil, errors.New("nil FabricIfaceParams")
+	}
+
+	fiList, err := n.Find(params.Interface)
+	if err != nil {
+		return nil, err
+	}
+
+	if params.Domain != "" {
+		fiList = filterDomain(params.Domain, fiList)
+		if len(fiList) == 0 {
+			return nil, errors.Errorf("fabric interface %q doesn't have requested domain %q",
+				params.Interface, params.Domain)
+		}
+	}
+
+	if params.Provider != "" {
+		fiList = filterProvider(params.Provider, fiList)
+		if len(fiList) == 0 {
+			return nil, errors.Errorf("fabric interface %q doesn't support provider %q",
+				params.Interface, params.Provider)
+		}
+	}
+
+	return fiList, nil
+}
+
+func filterDomain(domain string, fiList []*FabricInterface) []*FabricInterface {
+	result := make([]*FabricInterface, 0, len(fiList))
```

Comment:
Nitpicking, but we are probably going to do some overprovisioning as all the elements of the `filist` will probably not be added to result.

Reply:
Fair point, although we're not overprovisioning by much, as each network interface has 2-3 domains at most.